index.bs

```diff
@@ -3331,6 +3331,207 @@ be the same.
+    effectively replicate the overload logic. If the two overloads have little to no shared parts,
+    it is better to leave overload resolution to the IDL mechanism.
+
+    <div algorithm="execute supports() with optional argument">
+
+        The <code class="idl">supports(|propertyOrConditionText|, |value|)</code> method, when
+        called, must run these steps:
+
+        1.  If |value| is given, then:
+            1.  Let <var ignore>property</var> be |propertyOrConditionText|.
+            1.  …
+        1.  Otherwise:
+            1.  Let <var ignore>conditionText</var> be |propertyOrConditionText|.
+            1.  …
+
+    </div>
+
+*   If the operation accepts multiple types for multiple arguments with no coupling between types of
+    different arguments, [=union types=] can sometimes be the only viable solution.
+
+    <pre highlight="webidl">
+        typedef (long long or DOMString or CalculatableInterface) SupportedArgument;
+        interface A {
+          void add(SupportedArgument operand1, SupportedArgument operand2);
+        };
+    </pre>
+
+    For the <code class="idl">add()</code> operation above, to specify it using
+    [=overloaded|overloads=] would require
+
+    <pre highlight="webidl">
+        interface A {
+          void add(long long operand1, long long operand2);
+          void add(long long operand1, DOMString operand2);
+          void add(long long operand1, CalculatableInterface operand2);
+          void add(DOMString operand1, long long operand2);
+          void add(DOMString operand1, DOMString operand2);
+          void add(DOMString operand1, CalculatableInterface operand2);
+          void add(CalculatableInterface operand1, long long operand2);
+          void add(CalculatableInterface operand1, DOMString operand2);
+          void add(CalculatableInterface operand1, CalculatableInterface operand2);
+        };
+    </pre>
+
+    and nine times the corresponding prose!
+
+*   Specification authors are encouraged to treat missing argument and <emu-val>undefined</emu-val>
+    argument the same way in the ECMAScript language binding.
+
+    Given the following IDL fragment:
+
+    <pre highlight="webidl">
+        interface A {
+          void foo();
+          void foo(Node? arg);
+        };
+    </pre>
+
+    Using the ECMAScript language binding, calling <code>foo(undefined)</code> and
+    <code>foo(null)</code> would both run the steps corresponding to the <code
+    class="idl">foo(|arg|)</code> operation, with |arg| set to null, while <code>foo()</code> alone
+    would go to the first overload. This may be a surprising behavior for many API users. Instead,
+    it is recommended to use an optional argument, which would categorize both <code>foo()</code>
+    and <code>foo(undefined)</code> as "|arg| is not present".
+
+    <pre highlight="webidl">
+        interface A {
+          void foo(optional Node? arg);
+        };
+    </pre>
```

Comment:
Maybe a conclusion of "In general, optionality is best expressed using the optional keyword, and not using overloads".